File: src/main/java/com/br/gabrielmartins/syntri/api/actionbar/ActionBarUtil.java
package com.br.gabrielmartins.syntri.api.actionbar;

import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

import java.lang.reflect.Constructor;
import java.lang.reflect.Method;

public class ActionBarUtil {

    public static void send(Player player, String message) {
        try {
            Class<?> chatComponentText = getNMSClass("ChatComponentText");
            Class<?> iChatBaseComponent = getNMSClass("IChatBaseComponent");
            Class<?> packetPlayOutChat = getNMSClass("PacketPlayOutChat");

            Constructor<?> chatConstructor = chatComponentText.getConstructor(String.class);
            Object chatComponent = chatConstructor.newInstance(message);

            Object packet;
            try {
                packet = packetPlayOutChat.getConstructor(iChatBaseComponent, byte.class).newInstance(chatComponent, (byte) 2);
            } catch (NoSuchMethodException e) {
                Class<?> chatMessageType = getNMSClass("ChatMessageType");
                Object gameInfo = chatMessageType.getEnumConstants()[2];
                packet = packetPlayOutChat.getConstructor(iChatBaseComponent, chatMessageType).newInstance(chatComponent, gameInfo);
            }

            Object handle = player.getClass().getMethod("getHandle").invoke(player);
            Object connection = handle.getClass().getField("playerConnection").get(handle);
            Method sendPacket = connection.getClass().getMethod("sendPacket", getNMSClass("Packet"));
            sendPacket.invoke(connection, packet);

        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private static Class<?> getNMSClass(String name) throws ClassNotFoundException {
        String version = Bukkit.getServer().getClass().getPackage().getName().split("\\.")[3];
        return Class.forName("net.minecraft.server." + version + "." + name);
    }
}

File: syntri-engine/src/main/java/com/br/gabrielmartins/engine/api/actionbar/ActionBarUtil.kt
package com.br.gabrielmartins.commons.api.actionbar

import org.bukkit.Bukkit
import org.bukkit.entity.Player

object ActionBarUtil {

    fun send(player: Player, message: String) {
        try {
            val chatComponentText = getNMSClass("ChatComponentText")
            val iChatBaseComponent = getNMSClass("IChatBaseComponent")
            val packetPlayOutChat = getNMSClass("PacketPlayOutChat")

            val chatConstructor = chatComponentText.getConstructor(String::class.java)
            val chatComponent = chatConstructor.newInstance(message)

            val packet = try {
                packetPlayOutChat.getConstructor(iChatBaseComponent, Byte::class.java)
                    .newInstance(chatComponent, 2.toByte())
            } catch (e: NoSuchMethodException) {
                val chatMessageType = getNMSClass("ChatMessageType")
                val gameInfo = chatMessageType.enumConstants[2]
                packetPlayOutChat.getConstructor(iChatBaseComponent, chatMessageType)
                    .newInstance(chatComponent, gameInfo)
            }

            val handle = player.javaClass.getMethod("getHandle").invoke(player)
            val connection = handle.javaClass.getField("playerConnection").get(handle)
            val sendPacket = connection.javaClass.getMethod("sendPacket", getNMSClass("Packet"))
            sendPacket.invoke(connection, packet)

        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    private fun getNMSClass(name: String): Class<*> {
        val version = Bukkit.getServer().javaClass.packageName.split(".")[3]
        return Class.forName("net.minecraft.server.\$version.\$name")
    }
}